client/material/button.tsx

```diff
@@ -272,12 +246,54 @@ export function useButtonState({
       onMouseDown: handleMouseDown,
       onMouseEnter: handleMouseEnter,
       onMouseLeave: handleMouseLeave,
+      onKeyDown: handleKeyDown,
+      onKeyUp: handleKeyUp,
       $focused: focused,
     },
     rippleRef,
   ]
 }
 
+export interface HotkeyProp {
+  keyCode: number
+  altKey?: boolean
+  shiftKey?: boolean
+  ctrlKey?: boolean
+}
+
+export interface ButtonHotkeyProps {
+  /** The reference to the button that should be pressed programmatically. */
+  ref?: React.MutableRefObject<HTMLButtonElement | undefined>
```

Comment:
This being optional doesn't make any sense. Hotkeys don't work if you don't have a ref?